system_tests/nodeinterface_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021-2022, Offchain Labs, Inc.
+// For license information, see https://github.com/nitro/blob/master/LICENSE
+
+package arbtest
+
+import (
+	"context"
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/offchainlabs/nitro/arbos/util"
+	"github.com/offchainlabs/nitro/solgen/go/node_interfacegen"
+)
+
+func getL1BlockNum(t *testing.T, ctx context.Context, client *ethclient.Client, l2BlockNum uint64) uint64 {
+	header, err := client.HeaderByNumber(ctx, big.NewInt(int64(l2BlockNum)))
+	Require(t, err)
+	l1BlockNum := types.DeserializeHeaderExtraInformation(header).L1BlockNumber
+	return l1BlockNum
+}
```

Comment:
Why do we duplicate this function in test? Is it just for convenience so that we don't have to handle error it returns ? If so we should at least mark this as test helper, but this isn't really a test helper (that does setup/cleanup for tests) bur rather assertion helper which is not considered idiomatic in Go [0].
I suggest you use the real function you already have implemented in nodeinterface package, and handle errors.


[0] https://google.github.io/styleguide/go/best-practices#leave-testing-to-the-test-function

Reply:
This function is used in test in the process of first generating the true values (i.e the l2 blockranges wrt to l1 block numbers) and then use that to verify if `L2BlockRangeForL1` returns the correct range.

Reply:
Not sure I follow, this (currently `getBlockL1Num` anonymous function) does the same as `blockL1Num()` no ? Why can't we use that function? Am I missing something ?

If the intent here is to test that function as well, that can be a separate test.

Reply:
`blockL1Num()` function is private to nodeinterface though, like it is unexported and cannot be used directly here, we can export it but I am not sure

Reply:
Exporting it sounds good to me.

Speaking of needing to export functions just for tests, we should have more unit tests rather than having pretty much most of the tests in arbtest package, but that's outside of the scope of this pr.

Reply:
Cool